src/OpenLoco/src/Paint/Paint.cpp

```diff
@@ -1172,4 +1172,168 @@ namespace OpenLoco::Paint
+            }
+            else
+            {
+                lastEntry->children = entry;
+            }
+            auto* lastChild = entry;
+            for (; lastChild != nullptr; lastChild = lastChild->children)
+                ;
+            lastEntry = lastChild;
+        }
+
+        std::fill(std::begin(_E400D0), std::end(_E400D0), nullptr);
+
+        if (routeEntry == nullptr)
+        {
+            return;
+        }
+
+        auto minMaxBounds = routeEntry->bounds;
+        switch (getRotation())
+        {
+            case 0:
+                for (auto* child = routeEntry->children; child != nullptr; child = child->children)
+                {
+                    minMaxBounds.x = std::max(minMaxBounds.x, child->bounds.x);
+                    minMaxBounds.y = std::max(minMaxBounds.y, child->bounds.y);
+                    minMaxBounds.xEnd = std::min(minMaxBounds.xEnd, child->bounds.xEnd);
+                    minMaxBounds.yEnd = std::min(minMaxBounds.yEnd, child->bounds.yEnd);
+                }
+                break;
+            case 1:
+                for (auto* child = routeEntry->children; child != nullptr; child = child->children)
+                {
+                    minMaxBounds.x = std::min(minMaxBounds.x, child->bounds.x);
+                    minMaxBounds.y = std::max(minMaxBounds.y, child->bounds.y);
+                    minMaxBounds.xEnd = std::max(minMaxBounds.xEnd, child->bounds.xEnd);
+                    minMaxBounds.yEnd = std::min(minMaxBounds.yEnd, child->bounds.yEnd);
+                }
+                break;
+            case 2:
+                for (auto* child = routeEntry->children; child != nullptr; child = child->children)
+                {
+                    minMaxBounds.x = std::min(minMaxBounds.x, child->bounds.x);
+                    minMaxBounds.y = std::min(minMaxBounds.y, child->bounds.y);
+                    minMaxBounds.xEnd = std::max(minMaxBounds.xEnd, child->bounds.xEnd);
+                    minMaxBounds.yEnd = std::max(minMaxBounds.yEnd, child->bounds.yEnd);
+                }
+                break;
+            case 3:
+                for (auto* child = routeEntry->children; child != nullptr; child = child->children)
+                {
+                    minMaxBounds.x = std::max(minMaxBounds.x, child->bounds.x);
+                    minMaxBounds.y = std::min(minMaxBounds.y, child->bounds.y);
+                    minMaxBounds.xEnd = std::min(minMaxBounds.xEnd, child->bounds.xEnd);
+                    minMaxBounds.yEnd = std::max(minMaxBounds.yEnd, child->bounds.yEnd);
+                }
+                break;
+        }
+
+        routeEntry->bounds.x = minMaxBounds.x;
+        routeEntry->bounds.y = minMaxBounds.y;
+        routeEntry->bounds.xEnd = minMaxBounds.xEnd;
+        routeEntry->bounds.yEnd = minMaxBounds.yEnd;
+
+        addPSToQuadrant(*routeEntry);
+    }
+
+    // 0x0045CC1B
+    void PaintSession::finaliseTrackRoadAdditionsOrdering()
```

Comment:
these two are very similar should just merge them into one common function.

Reply:
Looks like only the destination addresses differ (`_E400D0` vs `_E400E4`). Is this something you'd like to parametrise out in this PR?